spring-cloud-kubernetes-fabric8-config/src/test/java/org/springframework/cloud/kubernetes/fabric8/config/Fabric8ConfigUtilsTests.java

```diff
@@ -78,4 +88,251 @@ void testNamespaceResolutionFailed() {
+		client.secrets().inNamespace("spring-k8s").create(new SecretBuilder()
+				.withMetadata(new ObjectMetaBuilder().withName("my-secret").withLabels(Map.of("color", "pink")).build())
+				.addToData(Map.of("property", Base64.getEncoder().encodeToString("value".getBytes()))).build());
+
+		Map.Entry<Set<String>, Map<String, Object>> result = Fabric8ConfigUtils.secretsDataByLabels(client,
+				"spring-k8s", Map.of("color", "pink"), new MockEnvironment(), Set.of(),
+				SourceDataEntriesProcessor::processAllEntries);
+		Assertions.assertEquals(Set.of("my-secret"), result.getKey());
+		Assertions.assertEquals(Map.of("property", "value"), result.getValue());
+	}
+
+	// secret "my-secret" is deployed with label {color:pink}; we search for it by same
+	// label and find it. This secret contains a single .yaml property, as such
+	// it gets some special treatment.
+	@Test
+	void testSecretDataByLabelsSecretFoundWithPropertyFile() {
+		client.secrets().inNamespace("spring-k8s").create(new SecretBuilder()
+				.withMetadata(new ObjectMetaBuilder().withName("my-secret").withLabels(Map.of("color", "pink")).build())
+				.addToData(Map.of("application.yaml", Base64.getEncoder().encodeToString("key1: value1".getBytes())))
+				.build());
+
+		Map.Entry<Set<String>, Map<String, Object>> result = Fabric8ConfigUtils.secretsDataByLabels(client,
+				"spring-k8s", Map.of("color", "pink"), new MockEnvironment(), Set.of(),
+				SourceDataEntriesProcessor::processAllEntries);
+		Assertions.assertEquals(Set.of("my-secret"), result.getKey());
+		Assertions.assertEquals(Map.of("key1", "value1"), result.getValue());
+	}
+
+	// secrets "my-secret" and "my-secret-2" are deployed with label {color:pink};
+	// we search for them by same label and find them.
+	@Test
+	void testSecretDataByLabelsTwoSecretsFound() {
+		client.secrets().inNamespace("spring-k8s").create(new SecretBuilder()
+				.withMetadata(new ObjectMetaBuilder().withName("my-secret").withLabels(Map.of("color", "pink")).build())
+				.addToData(Map.of("property", Base64.getEncoder().encodeToString("value".getBytes()))).build());
+
+		client.secrets().inNamespace("spring-k8s").create(new SecretBuilder()
+				.withMetadata(
+						new ObjectMetaBuilder().withName("my-secret-2").withLabels(Map.of("color", "pink")).build())
+				.addToData(Map.of("property-2", Base64.getEncoder().encodeToString("value-2".getBytes()))).build());
+
+		Map.Entry<Set<String>, Map<String, Object>> result = Fabric8ConfigUtils.secretsDataByLabels(client,
+				"spring-k8s", Map.of("color", "pink"), new MockEnvironment(), Set.of(),
+				SourceDataEntriesProcessor::processAllEntries);
+		Assertions.assertTrue(result.getKey().contains("my-secret"));
+		Assertions.assertTrue(result.getKey().contains("my-secret-2"));
+
+		Assertions.assertEquals(2, result.getValue().size());
+		Assertions.assertEquals("value", result.getValue().get("property"));
+		Assertions.assertEquals("value-2", result.getValue().get("property-2"));
+	}
+
+	/**
```

Comment:
I will need your take on this example too @ryanjbaxter - to tell me if this is a sane decision (I personally find it as such)